From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kristof Imeri <kristof@lanth.me>
Date: Tue, 27 Oct 2026 18:58:24 +0100
Subject: [PATCH 1/5] x11/selection: Free the size of INCR transfers

The property that starts an INCR transfer only holds the size of the
data that follows, and wasn't freed after reading it.
---
 src/x11/meta-x11-selection-input-stream.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/x11/meta-x11-selection-input-stream.c b/src/x11/meta-x11-selection-input-stream.c
index da7adb89e7..e616b82ed1 100644
--- a/src/x11/meta-x11-selection-input-stream.c
+++ b/src/x11/meta-x11-selection-input-stream.c
@@ -495,6 +495,7 @@ meta_x11_selection_input_stream_xevent (MetaX11SelectionInputStream *stream,
                     /* The remainder of the selection will come through PropertyNotify
                        events on xwindow */
                     priv->incr = TRUE;
+                    g_bytes_unref (bytes);
                     meta_x11_selection_input_stream_flush (stream);
                   }
                 else
